test/controllers/login_controller_test.rb

```diff
@@ -48,6 +48,31 @@ class LoginControllerTest < ActionController::TestCase
         after_count = User.count
         assert(after_count.must_equal(before_count + 1))
       end
+
+      it "should set flash notice when user role is nil" do
+
+        AclDao.any_instance.stubs(:get_role_and_daac).returns([nil, nil])
```

Comment:
Again, by stubbing get_role_and_daac, you are not testing the flow of control where CMR returns a response of a user not provisioned.  What happens if you create a EDL user id, try to access dashboard with that user id, and print out the response to the ACL query for your user that is not provisioned.   That response is what you want to mock.

Reply:
solved